### ISSUE TITLE
Prefer console.error over console.log for printing uncaught exceptions in h$errorMsg

### ISSUE THREAD

**Issue by iand675** [ghcjs/ghcjs#519]
Minor nitpick, but simply doing console.log seems insufficiently urgent for notifying developers of uncaught exceptions.

**Comment by luite** [comment]
Good point. there is a bit of abstraction going on, since the logging/reporting facilities depend on the platform that we're on.

Is there an existing JS library available that supports logging with various severity levels on our targets (mainly browser and nodejs, less functionality is acceptable for standalone JSC and SpiderMonkey)?

**Comment by iand675** [comment]
Well, node and browser both support these at the very least: `console.log`, `console.warn`, `console.error`. I spent some time trying to look for some logging wrappers that support all of those targets you mentioned, but nothing stood out immediately. SpiderMonkey supports `console.log`, and I've tried digging into the JSC source very briefly, but haven't found much.

So, I can think of a few possible ways of handling this:

1. Provide some sort of configuration options for the RTS, where a user can provide their own implementations for things like logging, e.g.

  ```
  // some sort of boot-time logging facility detection
  var baseLogger = console.log || (function () {})
  rts = {
    log: {
      info: console.log || baseLogger
      warn: console.warn || baseLogger
      error: console.error || baseLogger
    }
  }
  ```
2. Just bake in something like `(console.error || console.log)(thingToLog)` instead of making it fully configurable.
